subxt/src/backend/mod.rs

```diff
@@ -325,9 +325,395 @@ pub enum TransactionStatus<Hash> {
+            let entries = Data {
+                request: entries,
+                subscription: tx,
+            };
+            let rpc_client = RpcClient::new(MockRpcClientStorage {
+                data: Arc::new(Mutex::new(entries)),
+            });
+
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+            let response = backend
+                .storage_fetch_values(
+                    mock_data.iter().map(|x| x.0.clone().into()).collect(),
+                    crate::utils::H256::random(),
+                )
+                .await
+                .unwrap();
+
+            let response = response
+                .map(|x| x.unwrap())
+                .collect::<Vec<StorageResponse>>()
+                .await;
+            let expected: Vec<_> = mock_data
+                .iter()
+                .map(|(key, value)| storage_response(key, value))
+                .collect();
+            assert_eq!(expected, response)
+        }
+
+        #[tokio::test]
+        async fn storage_fetch_value() {
+            // Setup
+            let mock_data = ("ID1".to_owned(), "Data1".to_owned());
+            let mut entries = StorageEntry::new();
+            entries.push::<()>(
+                mock_data.0.clone().into(),
+                Err(RpcError::DisconnectedWillReconnect(
+                    "Reconnecting".to_string(),
+                )),
+            );
+            entries.push(mock_data.0.clone().into(), bytes(&mock_data.1));
+            let (_rx, tx) = tokio::sync::mpsc::channel(32);
+            let entries = Data {
+                request: entries,
+                subscription: tx,
+            };
+            let rpc_client = RpcClient::new(MockRpcClientStorage {
+                data: Arc::new(Mutex::new(entries)),
+            });
+
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+            let response = backend
+                .storage_fetch_value(mock_data.0.clone().into(), crate::utils::H256::random())
+                .await
+                .unwrap();
+
+            let response = response.unwrap();
+            assert_eq!(mock_data.1, String::from_utf8(response).unwrap())
+        }
+
+        #[tokio::test]
+        /// This test should cover following methods:
+        /// - `genesis_hash`
+        /// - `block_header`
+        /// - `block_body`
+        /// - `latest_finalized_block`
+        /// - `current_runtime_version`
+        /// - `current_runtime_version`
+        /// - `call`
+        /// The test covers them because they follow the simple pattern of:
+        /// ```no_run
+        ///  async fn THE_THING(&self) -> Result<T::Hash, Error> {
+        ///    retry(|| <DO THE THING> ).await
+        ///  }
+        /// ```
+        async fn simple_fetch() {
+            let hash = crate::utils::H256::random();
+            // Setup
+            let mut entries = StorageEntry::new();
```

Comment:
Picking hairs, it's bit strange with StorageEntry here to me it looks just mocked data for each rpc call/subscription. 

I would rather just to call it subscription and request data or something

Reply:
renamed to MockDataTable